train_and_save.py

```diff
@@ -0,0 +1,173 @@
+import matplotlib.pyplot as plt
```

Comment:
@guillaume-chevalier this is the final version. I will remove the old code in model.py, and migrate this code in the notebook. waiting for your review for this file only... I think it looks pretty neat with the new DeepLearningPipeline class :)